csrc/scheduler/matmul.cpp

```diff
@@ -350,50 +367,94 @@ void prologSwizzle(TensorView* shared_mem_tv, const MatmulParams& params) {
      *   7|          |          |
      *    +----------+----------+
      */
-
+    int64_t swizzle_period = n_rows / repeated_pattern_size;
+    // tile_size_y will be splitted by n_cols and then by swizzle_period,
+    // Recalculate swizzle_period if tile_size_y is smaller than the
+    // multiplication of these two split factors. If this happens,
+    // bank conflicts can't be fully removed.
+    if (tile_size_y < n_cols * swizzle_period) {
+      swizzle_period = tile_size_y / n_cols;
+      repeated_pattern_size = n_rows / swizzle_period;
+    }
+    // If the remaining part of tile_size_y is not divisible by swizzle_period,
+    // reduce swizzle_period by half until it is divisible. If this happens,
+    // bank conflicts can't be fully removed. e.g. tile_size_y = 96 in
+    // FusionAmpereMatmulTileCheck4warp_CUDA
+    while (tile_size_y / n_cols % swizzle_period) {
+      swizzle_period /= 2;
```

Comment:
Should `swizzle_period` be just `std::gcd(n_rows / repeated_pattern_size, tile_size_y / n_cols)`?

Reply:
accepted. This approach is more elegant.